sdk/samples/iot/generate_certificate.ps1

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+openssl ecparam -out device_ec_key.pem -name prime256v1 -genkey
+openssl req -new -days 365 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -config x509_config.cfg -subj "/CN=paho-sample-device1"
```

Comment:
Please check $LASTEXITCODE on each openssl call and doing a "throw" on any failure.  And any other exe we may call.

There's a real danger that if a ton of output here (either now or in future if you add more) then an `openssl` call failing could be buried in console history a developer would never see and they'd get some Frankenstein setup.  That would be worst of all worlds because it might even succeed but not be secure.

You can do a helper function to call openssl with arbitrary command like arguments.

There might be better ways to code this up but from our release scripts.

```script
function Invoke-IoTExecutable {

    param(
        [string]$Name = $(throw "Must provide executable name"),
        [string]$Command = $(throw "Must provide executable command string"),
        [string]$Path = (Get-Item .)
    )

    Push-Location $Path
    Invoke-Expression "$Name $Command"
    Pop-Location

    if ($LastExitCode -ne 0) {
        throw "'$Name' invocation failed with exit code ($LastExitCode)"
    }
}
```

Reply:
ahh thank you this is the kind of stuff I didn't know was available with Powershell. Will update